symengine/functions.cpp

```diff
@@ -42,6 +42,78 @@ extern umap_basic_basic inverse_cst;
 
 extern umap_basic_basic inverse_tct;
 
+Conjugate::Conjugate(const RCP<const Basic> &arg) : OneArgFunction(arg)
+{
+    SYMENGINE_ASSIGN_TYPEID()
+    SYMENGINE_ASSERT(is_canonical(arg))
+}
+
+bool Conjugate::is_canonical(const RCP<const Basic> &arg) const
+{
+    if (is_a_Number(*arg)) {
+        if (eq(*arg, *ComplexInf)) {
+            return true;
+        }
+        return false;
+    }
+    if (is_a<Constant>(*arg)) {
+        return false;
+    }
+    if (is_a<Mul>(*arg)) {
+        return false;
+    }
+    if (is_a<Pow>(*arg)) {
+        if (is_a<Integer>(*down_cast<const Pow &>(*arg).get_exp())) {
+            return false;
+        }
+    }
+    return true;
+}
+
+RCP<const Basic> Conjugate::create(const RCP<const Basic> &arg) const
+{
+    return conjugate(arg);
+}
+
+RCP<const Basic> conjugate(const RCP<const Basic> &arg)
+{
+    if (is_a_Number(*arg)) {
+        if (not down_cast<const Number &>(*arg).is_complex()) {
+            return arg;
+        }
+        if (is_a_Complex(*arg)) {
+            return down_cast<const ComplexBase &>(*arg).conjugate();
+        }
+    }
+    if (is_a<Constant>(*arg)) {
+        return arg;
+    }
+    if (is_a<Mul>(*arg)) {
+        RCP<const Basic> s = conjugate(down_cast<const Mul &>(*arg).get_coef());
+        map_basic_basic dict = down_cast<const Mul &>(*arg).get_dict();
+        map_basic_basic new_dict;
+        RCP<const Number> coef = one;
+        for (const auto &p : dict) {
+            if (is_a<Integer>(*p.second)) {
+                Mul::dict_add_term_new(outArg(coef), new_dict, p.second,
+                                       conjugate(p.first));
+            } else {
+                Mul::dict_add_term_new(outArg(coef), new_dict, one,
+                                       conjugate(pow(p.first, p.second)));
```

Comment:
Use `Mul::from_dict(one, {{p.first, p.second}})` instead of `pow(p.first, p.second)`